packages/react-router/src/useRenderedNavigationItems.tsx

```diff
@@ -66,8 +66,18 @@ export function useRenderedNavigationItems(
     renderSection: RenderSectionFunction
 ) {
     return useMemo(() => {
+        const clonedItems = [...navigationItems];
+
         // Highest priority is rendered first.
-        const sortedItems = [...navigationItems]
+        const sortedItems = clonedItems
+            .map(x => {
+                // Default an item priority to 0.
+                if (isUndefined(x.priority)) {
+                    x.priority = 0;
```

Comment:
It would be clearer to return a new object here instead of mutating the item directly.

It's a more common use case to mutate element in a for loop, or .forEach method. In a map, you expect the resulting array to have different values than the initial array. However in this case since we're modiying the src element, the input and output of the map function are the same.

So either this should be converted to a forEach function (which is less nice since you can't chain multiple operation like you are doing) or change this line to : 

```suggestion
                    return {
                        ...x,
                        priority: 0
                    }
```